### PR TITLE
Updating Data API and Euclid Documentation

### PR DESCRIPTION
**Pull request by IPadawans**
Updating the docs related to Euclid and Data API

### CODE REVIEW

File: sdk/elements/03-hydra-cli.md [hunk 1]
```diff
@@ -316,4 +316,81 @@ Host: :host
...
+
+The service is developed using `NodeJS`, and all necessary dependencies are installed on your remote instance during deployment.
+
+Running in the background with PM2, the service initiates checks at intervals specified in the configuration under the field: `check_healthy_interval_in_minutes`. It evaluates the health of the metagraph based on predefined and customizable `restart-conditions`, detailed in the [metagraph-monitoring-service](https://github.com/Constellation-Labs/metagraph-monitoring-service) repository. For example, if an unhealthy node is detected, the service triggers a restart.
+
+To restart a node or layer, the service first attempts to stop any running processes (referring to the layer). This operation requires sudo privileges without a password requirement (refer [to this](https://gcore.com/learning/how-to-disable-password-for-sudo-command/) document for instructions on setting up password-less sudo). In addition to terminating processes, log files from the node are moved to the `code/restart_logs` directory, which may also require sudo privileges.
+
+After these steps, the service restarts the node or layer and reintegrates it into the cluster.
```
Comment (AlexBrandes) [inline sdk/elements/03-hydra-cli.md:333]: I think this section needs to zoom out and describe what the service does at a high level. I don't think we need to mention PM2 or sudo privileges here. What does the service do? 

Something like... 
The service runs health checks on a regular interval and evaluates the metagraph cluster on a set of predefined or developer-created health criteria (restart-conditions). Restart conditions can target the whole cluster, a specific layer, or individual nodes to ensure that the metagraph is operating properly in each case. If an issue is detected, the service uses SSH to access the impacted node(s) and restart their process and rejoin them to the network.

File: sdk/elements/03-hydra-cli.md [hunk 2]
```diff
@@ -316,4 +316,81 @@ Host: :host
...
+-   `check_healthy_interval_in_minutes`: The interval, in minutes, for running the health check.
+
+NOTE: You must provide your SSH key file that has access to each node. It is recommended to place this under the `config` directory. Ensure that this file has access to the node and that the user you've provided also has sudo privileges without a password.
+
+### Customize Monitoring
+
+Learn how to customize your monitoring by checking the repositories:
+
+-   [metagraph-monitoring-service-package](https://github.com/Constellation-Labs/metagraph-monitoring-service-package)
+-   [metagraph-monitoring-service](https://github.com/Constellation-Labs/metagraph-monitoring-service-package)
```
Comment (AlexBrandes) [inline sdk/elements/03-hydra-cli.md:378]: What's the difference between these two repositories? I feel like these should be described separately.

File: sdk/frameworks/currency/01-overview.md [hunk 3]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
+
+-   `giter8`: A g8 template project that can be customized for your organization. This template can be built using [giter8](http://www.foundweekends.org/giter8/). For more details, visit the [project repository](https://github.com/Constellation-Labs/currency.g8).
+-   `metagraph-examples`: Explore ready-to-use examples of metagraphs. Check out all the examples [here](https://github.com/Constellation-Labs/metagraph-examples).
+
+:::note
+Installation requires pre-generated Tessellation dependencies. To generate these dependencies, run: For versions earlier than v2.3.0:
+
+```bash
+sbt shared/publishM2 kernel/publishM2 keytool/publishM2 sdk/publishM2 dagL1/publishM2 currencyL0/publishM2 currencyL1/publishM2
+```
```
Comment (AlexBrandes) [inline sdk/frameworks/currency/01-overview.md:29]: Should we even recommend any versions earlier than the latest?

**Review by AlexBrandes — CHANGES_REQUESTED** [review]
Apologies for so many comments - docs are hard. I think in general, we just need to zoom out from the technical details a little bit and make sure that the concepts are described clearly for newcomers that are starting with limited context.

- Would you check the Telemetry Dashboard section also? There's out of date info in there about how to start the old "monitoring" containers. 
- In the Developer Dashboard section, are we recommending the correct default ports? I think those changed in a previous version - should also update that repo if we haven't already.

File: sdk/frameworks/currency/01-overview.md [hunk 4]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
+
+For versions v2.3.0 or later:
+
+```bash
+sbt shared/publishM2 kernel/publishM2 keytool/publishM2 nodeShared/publishM2 dagL1/publishM2 currencyL0/publishM2 currencyL1/publishM2
+```
+:::
+
+## Installation using giter8
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:38]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 5]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
 ```
 
-Now you can install the template
+Then, install the template using the specified tag:
+
+```bash
+g8 Constellation-Labs/currency --tag "v2.3.3" #Using v2.3.3 as an example
+```
+
+## Installation using metagraph examples
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:52]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 6]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
 
 <intro-end />
 
-The Currency Framework allows you to quickly launch & customize an entire decentralized application with currency-related transactions supported natively and automatically. In contrast to closed loop systems like smart-contract platforms, greater flexibility in supporting conventional code is offered by the ability to directly modify any & all application level code. The functionalities associated with peer-to-peer currency transactions, chain data construction, and more are encapsulated in an easy-to-use framework to allow you to begin developing a decentralized application with it's own internal currency. 
+The Currency Framework provides a fast and customizable way to launch a decentralized application (dApp) with native support for currency-related transactions. Unlike closed-loop systems such as smart-contract platforms, this framework offers enhanced flexibility by allowing direct modifications to application-level code. It encapsulates functionalities necessary for peer-to-peer currency transactions, chain data construction, and more, in a user-friendly package. This enables developers to start creating a dApp with its own internal currency efficiently.
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:9]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 7]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
 
-This offers you the benefits of building and customizing your own application, with the support of a tested underlying framework saving you the effort involved in supporting conventional balance style transactions. Example entrypoints are supplied to make it quick and easy to launch your application immediately, which automatically supports a live, working token upon which your own features can be easily added.
+By using this framework, developers gain the advantage of building and customizing their own applications, supported by a proven underlying framework. This saves significant effort in handling conventional balance-style transactions. Example entry points are included to facilitate a quick and seamless application launch, which automatically supports a functioning, live token. Additional features can be easily integrated.
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:11]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 8]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
 
 :::info Euclid Development Environment
-It is recommended that you run this framework as part of a [Euclid Development Environment](/sdk/elements/dev-environment) project which will handle the installation and configuration of the project via the template described below, as well as provide a fully integrated minimal development environment. Manual installation instructions are provided below for completeness. 
+For optimal results, it is recommended to run this framework as part of a [Euclid Development Environment](https://chat.openai.com/sdk/elements/dev-environment) project. This environment manages the installation and configuration of the project as described below and provides an integrated minimal development setup. Manual installation instructions are also provided for completeness.
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:14]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 9]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
 :::
 
 ## Distribution
-The Currency Framework is distributed as a g8 template project that can be built with [giter8](http://www.foundweekends.org/giter8/) and customized to your organization. The template exposes the `Main` method for both the `L0` and `L1` applications which allows full customization of their functionality. 
 
-See the [project repo](https://github.com/Constellation-Labs/currency.g8) for further detail.
+The Currency Framework can be distributed in several ways, including:
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:19]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 10]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
-Now you can install the template
+Then, install the template using the specified tag:
+
+```bash
+g8 Constellation-Labs/currency --tag "v2.3.3" #Using v2.3.3 as an example
+```
+
+## Installation using metagraph examples
+
+Clone the [repository](https://github.com/Constellation-Labs/metagraph-examples) and select the desired example. Navigate to the example directory to proceed.
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:54]: Yeah, we can remove the manual suggestion

File: sdk/frameworks/currency/01-overview.md [hunk 11]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
+For data-l1:
 
-Compile and run the generated project. 
+```bash
+sbt dataL1/assembly
+```
 
+:::note
+It is highly recommended to use the [Euclid Development Environment](https://chat.openai.com/sdk/elements/dev-environment) for installation and compilation processes. Euclid simplifies the generation of dependencies and your metagraph JARs.
+:::
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:79]: Done

File: sdk/frameworks/currency/01-overview.md [hunk 12]
```diff
@@ -6,34 +6,74 @@ hide_table_of_contents: false
...
+```
+
+## Installation using metagraph examples
+
+Clone the [repository](https://github.com/Constellation-Labs/metagraph-examples) and select the desired example. Navigate to the example directory to proceed.
+
+## Compiling the Project
+
+After installing your project and the Tessellation dependencies, compile the project to generate your local JAR files. You can compile as follows:
+
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/01-overview.md:59]: Updated

File: sdk/frameworks/currency/02-data-api.md [hunk 13]
```diff
@@ -51,22 +51,59 @@ Data is submitted to an application through a single REST endpoint on data L1 no
 }
 ```
 
+The endpoint is designed to support multiple data types through a single access point, and these types are parsed within your application code. To accommodate different types of data, you should implement a lifecycle function named `signedDataEntityDecoder`. This function takes in your data and converts it to the `Signed[DataUpdate]` class, as defined in your application. For a practical example on how requests are parsed into the `Signed[DataUpdate]` class, refer to the [DOR Metagraph example](https://github.com/Constellation-Labs/dor-metagraph).
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/02-data-api.md:54]: Updated

File: sdk/frameworks/currency/02-data-api.md [hunk 14]
```diff
@@ -51,22 +51,59 @@ Data is submitted to an application through a single REST endpoint on data L1 no
...
+
+### OnChainState vs. CalculatedState
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/02-data-api.md:56]: Done

File: sdk/frameworks/currency/02-data-api.md [hunk 15]
```diff
@@ -51,22 +51,59 @@ Data is submitted to an application through a single REST endpoint on data L1 no
...
+A Data Application enables developers to define state schemas for their metagraph. Initial states are set in the `genesis` method of the `DataApplicationL0Service`, which can be overridden:
+
+The definition for `OnChainState` must extend the `DataOnChainState` class.
+The definition for `CalculatedState` must extend the `DataCalculatedState` class.
+For example, you can define states using map types as shown in the Scala code below:
 
-For example, you could define State using numeric types:
 ```scala
-class State(length: Long, width: Long) extends DataState
+class OnChainState(updates: List[Update]) extends DataOnChainState
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/02-data-api.md:74]: Updated

File: sdk/frameworks/currency/02-data-api.md [hunk 16]
```diff
@@ -51,22 +51,59 @@ Data is submitted to an application through a single REST endpoint on data L1 no
...
+  "value": {
+    // This type is defined by your application code
+  },
+  "proofs": [{
+    "id": "<public key>",
+    "signature": "<signature of data in value key above>"
+  }]
+}
+```
+Here one example of how to use it:
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/02-data-api.md:93]: Updated

File: sdk/frameworks/currency/02-data-api.md [hunk 17]
```diff
@@ -150,6 +203,44 @@ For example, serialize to/from a JSON update:
   }
 ```
 
+#### setCalculatedState
+This function updates the `calculatedState`. Below is an example of how to implement an in-memory update of `calculatedState`.
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/frameworks/currency/02-data-api.md:207]: No this is a separated function, I've updated

File: sdk/elements/03-hydra-cli.md [hunk 18]
```diff
@@ -316,4 +316,81 @@ Host: :host
 Public port: :your_port
 P2P port: :your_port
 Peer id: :peerId
-```
\ No newline at end of file
+```
+## Remote Monitoring
```
[the remark this answers is not in the document]
Comment (IPadawans) [reply sdk/elements/03-hydra-cli.md:320]: Moved

**Review by AlexBrandes — APPROVED** [review]

**Pull request merged**